components/setting_item_max.tsx

```diff
@@ -1,131 +1,53 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {FormattedMessage} from 'react-intl';
 
 import SaveButton from 'components/save_button';
 import Constants from 'utils/constants';
 import {isKeyPressed} from 'utils/utils';
 
-export default class SettingItemMax extends React.PureComponent {
+export type Props = {
+    inputs: ReactNode;
+    containerStyle?: string;
+    clientError?: string | React.ReactNode | null;
+    serverError?: {server_error_id: string; message: string} | string | null;
+    extraInfo?: string | JSX.Element | null;
```

Comment:
quite the same goes for `JSX.Element` which also has `string` in the union type already, so you can remove the extra `string` here